### PR TITLE
Batch reading

### PR DESCRIPTION
**Pull request by cosmo0920**

### CODE REVIEW

**Comment by cosmo0920** [comment]
This PR should be merged after long running test with benchmarking tool.
Please wait a minute to review this.

**Comment by cosmo0920** [comment]
### Previous memory and CPU usage

serial-reading-32bytes-message-result

### This patch memory and CPU usage

batch-reading-32bytes-message-result

### Conclusion

* CPU usage slightly down 3~5%
* Memory usage is steady

**Comment by cosmo0920** [comment]
I've run 500000 events benchmarking.
This benchmark indicates that a serious memory leak does not exist.

File: ext/winevt/winevt_query.c [hunk 1]
```diff
@@ -120,15 +121,24 @@ rb_winevt_query_set_timeout(VALUE self, VALUE timeout)
...
+  EVT_HANDLE hEvents[QUERY_ARRAY_SIZE];
   ULONG count;
+  DWORD status = ERROR_SUCCESS;
   struct WinevtQuery* winevtQuery;
 
   TypedData_Get_Struct(self, struct WinevtQuery, &rb_winevt_query_type, winevtQuery);
 
-  if (EvtNext(winevtQuery->query, 1, &event, INFINITE, 0, &count) != FALSE) {
-    winevtQuery->event = event;
+  if (!EvtNext(winevtQuery->query, QUERY_ARRAY_SIZE, hEvents, INFINITE, 0, &count) != FALSE) {
```
Comment (ashie) [inline ext/winevt/winevt_query.c]: Please, please remove the trailing `!= FALSE` :-1:

File: ext/winevt/winevt_query.c [hunk 2]
```diff
@@ -120,15 +121,24 @@ rb_winevt_query_set_timeout(VALUE self, VALUE timeout)
...
+    if (ERROR_NO_MORE_ITEMS != (status = GetLastError())) {
```
Comment (ashie) [inline ext/winevt/winevt_query.c]: I don't prefer doing `status = GetLastError()` in the conditional expression.
It's hard to distinguish compare & assignment for me.

File: ext/winevt/winevt_query.c (hunk 1 above)
Comment (cosmo0920) [reply ext/winevt/winevt_query.c]: Ouch! :confounded:

File: ext/winevt/winevt_query.c (hunk 2 above)
Comment (cosmo0920) [reply ext/winevt/winevt_query.c]: Fixed in https://github.com/fluent-plugins-nursery/winevt_c/pull/7/commits/f6ad1ed47526c5abcad74bb671bc07bffbba0955

**Comment by cosmo0920** [comment]
Thanks!

**Pull request merged**